pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/auth0/Auth0M2MRequest.java

```diff
@@ -0,0 +1,36 @@
+package org.broadinstitute.dsm.model.auth0;
+
+import lombok.Data;
+import lombok.NonNull;
+import org.apache.http.NameValuePair;
+import org.apache.http.message.BasicNameValuePair;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+@Data
+public class Auth0M2MRequest {
+    public List<NameValuePair> params;
+    private final String clientIdKey = "client_id";
+    private final String grantTypeKey = "grant_type";
+    private final String clientSecretKey = "client_secret";
+    private final String audienceKey = "audience";
+
+    public Auth0M2MRequest(@NonNull String clientId,@NonNull String grantType,@NonNull String clientSecret,@NonNull String audience, Map<String, String> claims, String audienceNameSpace){
+
+        //adding the request values with key name (audienceNameSpace + key) because auth0 requires claims to have a valid url as the name
+        this.params = new ArrayList<>();
+        for (String key: claims.keySet()){
+            String k = key;
+            if(key.indexOf(audienceNameSpace) == -1)
```

Comment:
Why are we looking up with audienceNameSpace inside the keySet loop iteration ? 
Do we meant to lookup audienceNameSpace + k ?

Reply:
This is because DSS needed claims that are added per `RULES` to the first token ( token obtained while doing the Google login) already has the namespace added to them (they are set as `audienceNameSpace + \ + claimName` in auth0 rule ), so I check to make sure I don't add the URL name space to them twice